discovery-provider/src/api/v1/models/activities.py

```diff
@@ -0,0 +1,38 @@
+from flask_restx import fields
+from flask_restx.fields import MarshallingError
+from flask_restx.marshalling import marshal
+from .common import ns
+from .tracks import track, track_full
+from .playlists import playlist_model, full_playlist_model
+
+class Item(fields.Raw):
+    def format(self, value):
+        try:
+            if value.get("track_id"):
+                return marshal(value, track)
+            if value.get("playlist_id"):
+                return marshal(value, playlist_model)
+        except:
+            raise MarshallingError("Unable to marshal as activity item")
+
+
+class FullItem(fields.Raw):
+    def format(self, value):
+        try:
+            if value.get("track_id"):
+                return marshal(value, track_full)
+            if value.get("playlist_id"):
+                return marshal(value, full_playlist_model)
+        except:
+            raise MarshallingError("Unable to marshal as activity item")
+
+
+activity_model = ns.model("activity", {
+    "timestamp": fields.String(allow_null=True),
+    "item": Item
```

Comment:
If we didn't put `is` in the object we could also put it at this level and leave tracks/playlists the same, but I think we agreed to put it in the item. just wanted to point that out though. This looks good!

Reply:
Yeah, I think you're right. This level is better

Reply:
Only problem then is that after the query returns, the API layer needs to check for a field like track_id or something. Wdyt?